### ISSUE TITLE
Support for Training Negative Samples

### ISSUE THREAD

**Issue by sphcn** [zylo117/Yet-Another-EfficientDet-Pytorch#12]
Hi, thanks for your code.
I have a question. What would be the good way to add support for **Negative Samples** or **Background Label** in training in this repository?

**Comment by zylo117** [comment]
there won't be, it doesn't have a class background, because it doesn't have a RPN that separate backgrond and foreground like RCNN. It selects the biggest class' logits to be the class of the anchors.

If you are asking how to deal with negative/positive anchors inbalance, you don't need to worry about that, [focal loss](https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch/blob/master/efficientdet/loss.py#L27) will do its job.

**Comment by sphcn** [comment]
I'm doing an object dectection task with a small dataset. I want to know how to deal with the images with no targets. Some images only have the backgrounds.  I don't want to discard the empty images.  Do you have any suggestions?
I'm a new hand in this field. Thanks a lot for your replies.

**Comment by zylo117** [comment]
It seems you have to discard them or the program would. Efficientdet is a one-stage anchor detection network, there is no background class at all. 

If you really want the model to be robust, you can apply augmentation, by patching the objects from the other images to the image with no target.

**Comment by sphcn** [comment]
Sounds like a solution.  Thanks anyway!
